var/spack/repos/builtin/packages/nvidia-nsight-systems/package.py

```diff
@@ -0,0 +1,113 @@
+_versions = {
+    "2024.6.1": {
+        "Linux-aarch64": (
+            "24700c28dfda9f95d4e93de218b86ab1ba0ee8b74cb61c3c581767296159c75c",
+            "https://developer.nvidia.com/downloads/assets/tools/secure/nsight-systems/2024_6/nsight-systems-2024.6.1-2024.6.1.90_3490548-0.aarch64.rpm",
+        ),
+        "Linux-x86_64": (
+            "dd4359a47ff3857395c55a0da483b64f5c0c3a1a2e57dd543a512dc3d2cd2674",
+            "https://developer.nvidia.com/downloads/assets/tools/secure/nsight-systems/2024_6/nsight-systems-2024.6.1-2024.6.1.90_3490548-0.x86_64.rpm",
+        ),
+    },
+    "2024.1.1": {
+        "Linux-aarch64": (
+            "41dc15ae128ef1de8e582b66bb465ac6bd67b9d20ef77fc70528b735d80fb3ec",
+            "https://developer.download.nvidia.com/devtools/repos/rhel8/arm64/nsight-systems-2024.1.1-2024.1.1.59_3380207-0.aarch64.rpm",
+        ),
+        "Linux-ppc64le": (
+            "8c98b511df1747c4c782430504ae6fa4b3fce6fa72623083a828fc0a1e11f1b8",
+            "https://developer.download.nvidia.com/devtools/repos/rhel8/ppc64le/nsight-systems-cli-2024.1.1-2024.1.1.59_3380207-0.ppc64le.rpm",
+        ),
+        "Linux-x86_64": (
+            "96f57548e0bd69cb02cd1fe8c70ed4a650636ecb3a5ea5ec490c8049adc2beb5",
+            "https://developer.download.nvidia.com/devtools/repos/rhel8/x86_64/nsight-systems-2024.1.1-2024.1.1.59_3380207-0.x86_64.rpm",
+        ),
+    },
+}
+
+
+class NvidiaNsightSystems(Package):
+    """NVIDIA Nsight™ Systems is a system-wide performance analysis tool designed
+    to visualize an application’s algorithms, identify the largest opportunities
+    to optimize, and tune to scale efficiently across any quantity or size of CPUs
+    and GPUs, from large servers to the smallest system on a chip"""
+
+    homepage = "https://developer.nvidia.com/nsight-systems"
+    url = "https://developer.download.nvidia.com/devtools/repos/"
+    maintainers("scothalverson")
+    license("NVIDIA Software License Agreement")
+
+    depends_on("libarchive programs='bsdtar'")
+
+    for ver, packages in _versions.items():
+        key = "{0}-{1}".format(platform.system(), platform.machine())
+        pkg = packages.get(key)
+        if pkg:
+            version(ver, sha256=pkg[0], url=pkg[1], expand=False)
+
+    def install(self, spec, prefix):
+        bsdtar = which("bsdtar")
+        rpm_file = glob(join_path(self.stage.source_path, "nsight-systems*.rpm"))[0]
+        params = ["-x", "-f", rpm_file]
+        ver = prefix.split("/")[-1].split("-")[-2]
+        bsdtar(*params)
+
+        arch = self.spec.target.family
+        if arch == "aarch64":
+            folders = ["documentation", "host-linux-armv8", "target-linux-sbsa-armv8"]
+        elif arch == "ppc64le":
+            folders = ["documentation", "host-linux-ppc64le", "target-linux-ppc64le"]
+        elif arch == "x86_64":
+            folders = ["documentation", "host-linux-x64", "target-linux-x64"]
```

Comment:
Add an `else:` fail.

Reply:
Is that really necessary?  There are no valid versions at all of the package for other architectures, you wouldn't even be able to get to this line.

Reply:
I think you are right, requesting a different arch would just make it fail an earlier level.